src/stdlib_math.fypp

```diff
@@ -1,21 +1,270 @@
+  !! If the number of elements is not specified, create an array with size 50. If n is a negative value,
+  !! return an array with size 0. If n = 1, return an array whose only element is base**end. If no base
+  !! is specified, logspace will default to using a base of 10
+  !!
+  !!([Specification](../page/specs/stdlib_math.html#logspace))
+  #!=========================================================
+  #!=     logspace(start, end)                              =
+  #!=========================================================
+  #:for k1, t1 in RC_KINDS_TYPES
+    #:set RName = rname("logspace", 1, t1, k1, "default")
+    module function ${RName}$(start, end) result(res)
+
+      ${t1}$, intent(in) :: start
+      ${t1}$, intent(in) :: end
+
+      ${t1}$ :: res(DEFAULT_LOGSPACE_LENGTH)
+
+    end function ${RName}$
+  #:endfor
+  #! Integer support
+  #:set RName = rname("logspace", 1, "integer(int32)", "int32", "default")
+    module function ${RName}$(start, end) result(res)
+
+      integer, intent(in) :: start
+      integer, intent(in) :: end
+
+      real(dp) :: res(DEFAULT_LOGSPACE_LENGTH)
+
+  end function ${RName}$
+
+  #!=========================================================
+  #!=     logspace(start, end, n)                           =
+  #!=========================================================
+  #:for k1, t1 in RC_KINDS_TYPES
+    #:set RName = rname("logspace", 1, t1, k1, "n")
+    module function ${RName}$(start, end, n) result(res)
+      ${t1}$, intent(in) :: start
+      ${t1}$, intent(in) :: end
+      integer, intent(in) :: n
+
+      ${t1}$ :: res(n)
+    end function ${RName}$
+  #:endfor
+  #! Integer support
+  #:set RName = rname("logspace", 1, "integer(int32)", "int32", "n")
+    module function ${RName}$(start, end, n) result(res)
+      integer, intent(in) :: start
+      integer, intent(in) :: end
+      integer, intent(in) :: n
+
+      real(dp) :: res(n)
+    end function ${RName}$
+
+  #!=========================================================
+  #!=     logspace(start, end, n, base)                     =
+  #!=========================================================
+  #! Need another function where base is not optional,
+  #! otherwise the compiler can not differentiate between
+  #! generic calls to logspace_n where a base is not present
+  #! ========================================================
+  #:for k1, t1 in REAL_KINDS_TYPES
+    ! Generate logarithmically spaced sequence from ${k1}$ base to the powers
+    ! of ${k1}$ start and end. [base^start, ... , base^end]
+    ! RName = ${RName}$
+    #:set RName = rname("logspace", 1, t1, k1, "n_rbase")
+    module function ${RName}$(start, end, n, base) result(res)
+      ${t1}$, intent(in) :: start
+      ${t1}$, intent(in) :: end
+      integer, intent(in) :: n
+      ${t1}$, intent(in) :: base
+      ! real(${k1}$) endpoints + real(${k1}$) base = real(${k1}$) result
```

Comment:
I don't understand this comment.

Reply:
This comment is a bookkeeping note that I made while developing the generic function templates. It indicates that when passed real endpoints and a real base, we expect to get a real result. 

Compare this to other combinations.

When we pass real endpoints + complex base, the result shall be complex
When we pass integer endpoints + integer base, the result shall be an integer
When we pass integer endpoints + real base, the result shall be real

Reply:
Great. Useful comment indeed. Could it be a bit more verbose, such that it becomes obvious for a reader?

Reply:
Sure it can. I'm gonna get some lunch and then I'll be back on to touch up the documentation a tad.